submariner/multiple-active-gateways.md

```diff
@@ -0,0 +1,100 @@
+# Multiple Active Gateways
+
+## Summary
+
+NOTE: The design details are still a work in progress, so this is currently a draft proposal.
+
+Submariner currently only allows a single gateway to be active at any one time in a given cluster.
+This enhancement proposes that there be an option to allow more than one gateway to be active in
+a given cluster to enhance both performance and resiliency. This feature is sometimes referred to as
+“active-active gateways”; however, the goal is to allow two or more actove gateways to be used.
+
+## Proposal
+
+The proposal is to optionally allow multiple gateways to be active at the same time in a given cluster.
+
+When only a single active gateway is used, all inter-cluster traffic must flow through that one gateway,
+which creates a performance bottleneck. Enabling more than one node to be used as a gateway can improve
```

Comment:
Maybe go in a little more about how this is a performance bottleneck specifically with encryption engines? I.e All encryption is locked to one core etc.

Reply:
Ok.

Reply:
It seems that the painful points in the current solution are performance bottleneck and slow failover time.

While the proposed solution addresses the performance bottleneck, it complicates the datapath handling both intra-cluster and inter-cluster. 

Are there other alternatives (e.g: setting more resources somehow to vpn pod, etc) to mitigate the performance bottleneck?

Do we know how critical is the failover timing? the maximum time for GW failover? 
Is it possible to reduce GW failover timing with Active/SB architecture?  (Maybe we can use VIP to represent the active GW? )

Reply:
Improving the recovery time isn't the main objective.  So, I think of it as more of a bonus.  I'm sure there is room for improvement in the current design/implementation without going to MAG.